plutus-wallet-api/ledger/Ledger/Value.hs

```diff
@@ -303,36 +317,29 @@ checkBinRel f l r =
 {-# INLINABLE geq #-}
 -- | Check whether one 'Value' is greater than or equal to another. See 'Value' for an explanation of how operations on 'Value's work.
 geq :: Value -> Value -> Bool
+-- If both are zero then checkBinRel will be vacuously true, but this is fine.
 geq = checkBinRel P.geq
 
 {-# INLINABLE gt #-}
 -- | Check whether one 'Value' is strictly greater than another. See 'Value' for an explanation of how operations on 'Value's work.
 gt :: Value -> Value -> Bool
-gt = checkBinRel P.gt
+-- If both are zero then checkBinRel will be vacuously true. So we have a special case where l is zero.
+gt l r = not (isZero l) `P.and` checkBinRel P.gt l r
```

Comment:
Why not 

```suggestion
gt l r = not (isZero l `P.and` isZero r) `P.and` checkBinRel P.gt l r
```

I think `zero > { "a" -> -1 }` should be true, if we assume the values at all absent keys to be 0. Same for `lt` below.

Reply:
Uh right, we have negative quantities.

Reply:
I'm not sure that we should have negative quantities. In the specification I've made it so that you're only allowed negative quantities when you're forging.  This seems like a small change that avoids potential pitfalls.